### PR TITLE
DOC-1772: Added doc for text_patterns_lookup

### PR DESCRIPTION
**Pull request by tiny-ben-tran**
Related Ticket: DOC-1772

Pre-checks:
- [x] Branch prefixed with `feature/6/` or `hotfix/6/`
- [x] ~`modules/ROOT/nav.adoc` has been updated (if applicable)~
- [x] Files have been included where required (if applicable)
- [x] ~Files removed have been deleted, not just excluded from the build (if applicable)~
- [x] ~(New product features only) Release Note added~

Review:
- [x] Documentation Team Lead has reviewed

### CODE REVIEW

File: modules/ROOT/partials/configuration/text_patterns_lookup.adoc [hunk 1]
```diff
@@ -0,0 +1,33 @@
+[[text_patterns_lookup]]
+== `+text_patterns_lookup+`
+
+This option allows you to specify a function that dynamically append additional text patterns to the xref:content-behavior-options.adoc#text_patterns[text_patterns]. The function takes a context object as an argument and should return a list of text patterns. The function will be called whenever the editor is attempting to do text pattern matching.
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/text_patterns_lookup.adoc:4]: No `you`, you shouldn't use pronouns per https://github.com/tinymce/tinymce-docs/blob/staging/docs-6/CONTRIBUTING.md#pronouns

File: modules/ROOT/partials/configuration/text_patterns_lookup.adoc [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+*Type:* `+Function+`
+
+=== Example: Using text_patterns_lookup
+
+[source,js]
+----
+tinymce.init({
+  selector: 'textarea', // change this value according to your HTML
+  text_patterns_lookup: (ctx) => {
+    const text = ctx.
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/text_patterns_lookup.adoc:15]: This isn't valid or is missing something (probably the latter).

File: modules/ROOT/partials/configuration/text_patterns_lookup.adoc [hunk 3]
```diff
@@ -0,0 +1,32 @@
+[[text_patterns_lookup]]
+== `+text_patterns_lookup+`
+
+This option allows specifying a function that dynamically appends additional text patterns to the xref:content-behavior-options.adoc#text_patterns[text_patterns]. The function takes a context object as an argument and should return a list of text patterns. The function will be called whenever the editor is attempting to do text pattern matching.
```
Comment (MitchCTiny) [inline modules/ROOT/partials/configuration/text_patterns_lookup.adoc:4]:
```suggestion
This option allows the specification of a function that dynamically appends additional text patterns to the default xref:content-behavior-options.adoc#text_patterns[text_patterns]. The function takes a context object as an argument and returns an array of text patterns. This function is called whenever the editor triggers text pattern matching.
```

File: modules/ROOT/partials/configuration/text_patterns_lookup.adoc [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    } else if (parentTag === 'p') {
+      return [
+        { start: '*', end: '*', format: 'bold' }
+      ];
+    } else {
+      return [
+        // ctx.text is the string from the start of the block to the cursor
+        { start: 'brb', replacement: ctx.text + ': Be Right Back' }
+      ];
+    }
```
Comment (MitchCTiny) [inline modules/ROOT/partials/configuration/text_patterns_lookup.adoc:29]: Should this also be an `else if`? Seems like returning this in all other cases isn't a great idea. Maybe just returning a list of text patterns when within a `p` tag is adequate? Just from what I can tell, but I don't really know the option.

Comment (spocke) [reply modules/ROOT/partials/configuration/text_patterns_lookup.adoc:29]: Maybe just have the last else be `return [];` showing how you should return things that doesn't match what you are looking for.

**Review by spocke — APPROVED** [review]

**Review by lorenzo-pomili — APPROVED** [review]

**Review by MitchCTiny — APPROVED** [review]

**Pull request merged**